### PR TITLE
Bug/responsive graphs 2

### PR DESCRIPTION
**Pull request by Ethan-Edmond**
# Description

[Explanation Video](https://www.loom.com/share/d960c3bd35a14efcbc5ca1e62b1e431b)
Makes the bar graph on the graphs page responsive, and makes it horizontal below 1100px of width. The structure of the bar graph has been simplified and recolored similarly to the line graph.

depends on bug/responsive-graphs-1

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [X] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [X] `npm test`

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] There are no merge conflicts

### CODE REVIEW

File: src/components/graphs/bargraph/BarGraph.js [hunk 1]
```diff
@@ -1,15 +1,57 @@
...
+          labelString: 'U.S. State',
+        },
+      },
+    ],
+  },
+  legend: {
+    ...graphOptions.legend,
+    position: 'top',
+  },
+};
```
Comment (lgarcia2689) [inline src/components/graphs/bargraph/BarGraph.js:47]: Adding this responsiveness is imperative for the the graphs visualization, great job

File: src/components/graphs/bargraph/BarGraph.js [hunk 2]
```diff
@@ -20,23 +62,50 @@ const def = {
 
 const BarGraph = ({ count }) => {
   const [barData, setBarData] = useState(def);
-  const [incidentCount, setIncidentCount] = useState(null);
+  const [horizBarData, setHorizBarData] = useState(def);
+  const [horizShowing, setHorizShowing] = useState(false);
```
Comment (lgarcia2689) [inline src/components/graphs/bargraph/BarGraph.js:67]: This looks awesome, really like the graphs came out

**Review by lgarcia2689 — APPROVED** [review]
Great job, was able to see all changes were needed both for usage and visualization of program

**Review by austin-carman — APPROVED** [review]
Great job making the graphs responsive and cleaning up the existing code too

File: src/components/graphs/bargraph/BarGraph.js [hunk 3]
```diff
@@ -20,23 +62,50 @@ const def = {
...
+    setHorizShowing(queryMatcher.matches);
+    const callback = ({ matches }) => {
+      setHorizShowing(matches);
+    };
+    queryMatcher.addListener(callback);
+
+    return () => {
+      queryMatcher.removeListener(callback);
+    };
+  }, []);
```
Comment (paulstgermain) [inline src/components/graphs/bargraph/BarGraph.js:79]: Considering this is the only place this is being used, it was reasonable to move it directly into the file it'd be used in - nice find!

**Review by paulstgermain — APPROVED** [review]
See my comments on `responsive graphs 2` - nice work explaining your work here as well!

File: src/components/graphs/bargraph/BarGraph.js [hunk 4]
```diff
@@ -1,15 +1,57 @@
...
+  legend: {
+    display: false,
+    position: 'left',
+    labels: {
+      boxWidth: 0,
+    },
+  },
+};
+
+const horizOptions = {
```
Comment (paulstgermain) [inline src/components/graphs/bargraph/BarGraph.js:18]: Forgot to mention - love that you introduced a horizontal bar graph per our conversation about better means to display this comprehensive data set!

**Pull request merged**